### PR TITLE
Major update with LADMAP implementation

### LINKED ISSUE #59: Extend input image to 3D
Rewrite the LADMAP routine to accept 3D images (3D DCT will be performed instead of 2D).

### PR DESCRIPTION
**Pull request by yfukai**
accepting 3D images to perform 3D DCT fitting, closes #59.

### CODE REVIEW

**Comment by yfukai** [comment]
Tests passing, I think it's now ready for review

**Comment by yfukai** [comment]
https://github.com/peng-lab/BaSiCPy/pull/54#pullrequestreview-1003692001
discussion on old impl.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -3,17 +3,20 @@ repos:
     rev: v4.1.0
     hooks:
       - id: check-docstring-first
+        exclude: experimental_code/
```
Comment (Nicholas-Schaub) [inline .pre-commit-config.yaml:6]: Can we remove references to `experimental_code` since we do not have this folder in the repository?

**Review by Nicholas-Schaub — APPROVED** [review]
This looks excellent. I think we will want to add more documentation and remove some commented code, but I don't see any real issues merging this. Good work!

**Pull request merged**